gobdd.go

```diff
@@ -116,6 +134,9 @@ type TestingT interface {
 	Run(name string, f func(t *testing.T)) bool
 }
 
+// TestingTKey is used to store reference to current *testing.T instance
+type TestingTKey struct{}
```

Comment:
this is what I'm not sure about. More handy would be just `ctx.T()`. @sagikazarmark WDYT?

Reply:
Why not `consider passing t as argument to scenario hooks`?

Reply:
yeah, we have 3 options here and I'm not sure which one is a better choice.

Reply:
> consider passing t as argument to scenario hooks

it's an API change

Reply:
I'd go with the one that we already have in place. Having T in both the context and as a parameter is redundant.

Reply:
> handy would be just `ctx.T()`

It might not be necessary in the future if we start passing `t` to hooks functions in v2.

Reply:
@sagikazarmark I am trying to find a long term solution here. I would rather change the API for hooks functions in v2 and keep the usage of `TestingT` as not encouraged until we do such change. The API of `gobbd.Context` should be simple and not mix concerns, hence I am not in favor of `ctx.T()`.

Reply:
> Having T in both the context and as a parameter is redundant.

Hook functions are not accepting `T` as parameter atm.

Reply:
I'm not 100% sure I understand which you are advocating.

I think testing.T should be added to hook functions as well. I believe we can do that in a backward compatible manner (see the relocation of context package).

Reply:
Honestly, I don't know how to change the type signature in backward compatible manner. If you can do it, then I would prefer passing the `t` as argument.

Reply:
@sagikazarmark @bkielbasa can we merge the current solution for v1 and change the hook function signature in v2?

Reply:
```suggestion
type TestingT struct{}
```

I think the word `Key` here is redundant.

Reply:
@bkielbasa `TestingT` is already defined as interface.